qcodes/instrument_drivers/Keysight/N51x1.py

```diff
@@ -58,6 +58,11 @@ def __init__(self, name: str, address: str, min_power: int = -144, max_power: in
                            unit='rad'
                            )
 
+        self.add_parameter('auto_freq_ref',
+                           get_cmd=':ROSC:SOUR:AUTO?',
+                           set_cmd=':ROSC:SOUR:AUTO {}',
+                           val_mapping={'on': 1, 'off': 0})
```

Comment:
It might make sense to use [create_on_off_val_mapping](https://qcodes.github.io/Qcodes/api/parameters/index.html?highlight=create_on_off#qcodes.parameters.create_on_off_val_mapping)

Reply:
Definitely. I wasn't aware of this function, and actually it can be added to the `rf_output` parameter as well.

Reply:
Sorry missed the last part. It's explained here https://qcodes.github.io/Qcodes/examples/writing_drivers/Creating-Instrument-Drivers.html#On/Off-parameters but perhaps that should have a more complete example